### PR TITLE
Use dynamic version of py-shiny for deploy tests

### PR DESCRIPTION
**Pull request by karangattu**
When we used `git+https://github.com/posit-dev/py-shiny.git` in the requirements.txt file for the deploy tests, `connect` would cache the version of py-shiny. The `connect` team suggested to use `hash` instead.
This commit dynamically fetches the hash of py-shiny `main` and adds to each of the `requirements.txt` for deploy tests.

Deleted the older mentions of `py-shiny`

Tested locally with one deploy test deploying to connect and it worked fine.

### CODE REVIEW

File: tests/playwright/utils/deploy_utils.py [hunk 1]
```diff
@@ -100,3 +100,12 @@ def deploy(location: str, app_name: str, app_file_path: str) -> str:
     else:
         raise ValueError("Unknown deploy location. Cannot deploy.")
     return url
+
+
+# Since connect parses python packages, we need to get latest version of shiny on main
+def tweak_requirements_txt(app_file_path: str) -> None:
+    requirements_file_path = os.path.join(app_file_path, "requirements.txt")
+    git_cmd = subprocess.run(["git", "rev-parse", "main"], stdout=subprocess.PIPE)
```
Comment (wch) [inline tests/playwright/utils/deploy_utils.py:108]: I think we don't want to base this off of `main` -- for example, i fthere's a PR, we want to use that branch. I think we can just use `HEAD` here.

```suggestion
    git_cmd = subprocess.run(["git", "rev-parse", "HEAD"], stdout=subprocess.PIPE)
```

File: tests/playwright/utils/deploy_utils.py [hunk 2]
```diff
@@ -100,3 +100,12 @@ def deploy(location: str, app_name: str, app_file_path: str) -> str:
     else:
         raise ValueError("Unknown deploy location. Cannot deploy.")
     return url
+
+
+# Since connect parses python packages, we need to get latest version of shiny on main
+def tweak_requirements_txt(app_file_path: str) -> None:
```
Comment (karangattu) [inline tests/playwright/utils/deploy_utils.py:106]: rename to `write_requirements_txt`

File: tests/playwright/utils/deploy_utils.py [hunk 3]
```diff
@@ -100,3 +100,12 @@ def deploy(location: str, app_name: str, app_file_path: str) -> str:
...
+    requirements_file_path = os.path.join(app_file_path, "requirements.txt")
```
Comment (karangattu) [inline tests/playwright/utils/deploy_utils.py:107]: Should read from app_requirements.txt in the app path

File: tests/playwright/utils/deploy_utils.py [hunk 4]
```diff
@@ -100,3 +100,12 @@ def deploy(location: str, app_name: str, app_file_path: str) -> str:
...
+    git_cmd = subprocess.run(["git", "rev-parse", "HEAD"], stdout=subprocess.PIPE)
+    git_hash = git_cmd.stdout.decode("utf-8").strip()
+    with open(requirements_file_path, "a") as f:
```
Comment (karangattu) [inline tests/playwright/utils/deploy_utils.py:110]: Write to `requirements.txt` in the app path

Comment (karangattu) [reply tests/playwright/utils/deploy_utils.py:110]: Make sure you add requirements.txt to `gitgignore`

**Review by schloerke — APPROVED** [review]

**Pull request merged**